lib/okta/jwk.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Okta.Jwk do
+  def fetch_jwk(%{"kid" => key_id}, payload) do
+    with {:ok, jwks_uri} <- fetch_jwks_uri(payload),
+         {:ok, jwks_response} <- HTTPoison.get(jwks_uri),
+         {:ok, decoded_body} <- Jason.decode(jwks_response.body),
+         {:ok, keys} <- Map.fetch(decoded_body, "keys") do
+      {
+        :ok,
+        Enum.find(keys, fn key -> Map.fetch!(key, "kid") == key_id end)
+      }
+    else
+      {:error, message} -> {:error, message}
+      message -> {:error, message}
+    end
+  end
+
+  def fetch_jwk(_, _) do
+    {:error, :invalid_token_header}
+  end
+
+  defp fetch_jwks_uri(%{"cid" => client_id, "iss" => issuer}) do
+    with {:ok, metadata_response} <-
+           HTTPoison.get(
+             "#{issuer}/.well-known/oauth-authorization-server?client_id=#{client_id}"
+           ),
```

Comment:
TODO: create an HTTP Client implementation instead of using HTTPoison directly, so that users may specify their preferred client, such as hackney